chargeio/infrastructure/Requestor.cs

```diff
@@ -0,0 +1,117 @@
+                }
+            }
+        }
+
+        private static string Put(string url, string putData, string contentType, string secretKey)
+        {
+            using(HttpClient client = new HttpClient()) {
+                try {
+                    var byteArray = Encoding.ASCII.GetBytes(Configuration.SecretKey + ":");
+                    client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Basic", Convert.ToBase64String(byteArray));
+                    client.Timeout = TimeSpan.FromMilliseconds(Configuration.HttpTimeout);
+                    client.DefaultRequestHeaders.Add("User-Agent", GetUserAgent());
+                    var result = client.PutAsync(url, new StringContent(putData, Encoding.UTF8, contentType));
+                    return result.Result.Content.ReadAsStringAsync().Result;
+                } catch (Exception e) {
+                    Console.WriteLine(e);
+                    return null;
+                }
+            }
+        }
+
+        private static string Post(string url, string putData, string contentType, string secretKey)
+        {
+            using(HttpClient client = new HttpClient()) {
+                try {
+                    var byteArray = Encoding.ASCII.GetBytes(Configuration.SecretKey + ":");
+                    client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Basic", Convert.ToBase64String(byteArray));
+                    client.Timeout = TimeSpan.FromMilliseconds(Configuration.HttpTimeout);
+                    client.DefaultRequestHeaders.Add("User-Agent", GetUserAgent());
+                    var result = client.PostAsync(url, new StringContent(putData, Encoding.UTF8, contentType));
+                    return result.Result.Content.ReadAsStringAsync().Result;
+                } catch (Exception e) {
+                    Console.WriteLine(e);
+                    return null;
+                }
+            }
+        }
+
+        private static string Delete(string url)
+        {
+            using(HttpClient client = new HttpClient()) {
+                try {
+                    var byteArray = Encoding.ASCII.GetBytes(Configuration.SecretKey + ":");
+                    client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Basic", Convert.ToBase64String(byteArray));
+                    client.Timeout = TimeSpan.FromMilliseconds(Configuration.HttpTimeout);
+                    client.DefaultRequestHeaders.Add("User-Agent", GetUserAgent());
+                    var result = client.DeleteAsync(url);
+                    return result.Result.Content.ReadAsStringAsync().Result;
+                } catch (Exception e) {
+                    Console.WriteLine(e);
+                    return null;
+                }
+            }
+        }
+
+        private static string GetUserAgent()
+        {
+            return "ChargeIO .net Client";

```

Comment:
Fixed in bd0da1b2eb71687d08f46125a729a2cd7e18a196

Unintended change. My mistake. Had to update the way that we were retrieving the assembly version to make sure that it was correct in all cases.